### PR TITLE
[Feature][#67] Adds tslint autofix as a flag to ForkTsCheckerWebpackPlugin plugin

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Adds `tslintAutoFix` as a flag to `ForkTsCheckerWebpackPlugin` plugin in webpack config.
- Set the default to false for backward compatibility.
- Updates Vue integration test based on changes to `IncrementalChecker` class update.

Addresses #67

### CODE REVIEW

**Comment by ajainarayanan** [comment]
@johnnyreilly Changes for adding auto fix as a flag. Let me know if I had missed anything.

**Comment by johnnyreilly** [comment]
Thanks for this! @piotr-oles what's your feeling on merging this? It's not a feature that I would use but clearly there's some demand out there for this. 

I know you expressed reservations so I don't want to merge unless I know you're okay with this!

**Comment by piotr-oles** [comment]
Thank you @ajainarayanan for your work! I was afraid that it will add complexity to the plugin but looks like it doesn't. Please add an integration test for tslintAutoFix: true case and update README.md :)

**Comment by ajainarayanan** [comment]
@piotr-oles @johnnyreilly This is was a tricky change. Have taken an approach that shouldn't affect the other tests. 

~There is still one problem with `should fix linting errors with tslintAutofix flag set to true` test. It lints other files as well even though I am explicitly passing in the entry point. Don't know what I am missing but right now my brain is fried and can't think of a reason. Will update the pull request when I have sometime. If you happen to identify the reason as to why it updates other files please do point out and I will fix it.~ 

UPDATE:
This is fixed now.

File: package.json [hunk 1]
```diff
@@ -110,7 +110,7 @@
         "eslint --fix"
       ],
       "*.ts": [
-        "tslint --fix"
+        "tslint --fix --exclude test/integration/project/lintingError.ts"
       ],
```
Comment (ajainarayanan) [inline package.json]: Sorry this is not needed. Will remove.

File: test/integration/index.spec.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+// @ts-nocheck
+var fs = require('fs');
```
Comment (ajainarayanan) [inline test/integration/index.spec.js:1]: This as well. Will remove

**Comment by johnnyreilly** [comment]
@ajainarayanan could you merge in the upstream changes please?

**Comment by ajainarayanan** [comment]
@johnnyreilly Have updated.

File: test/integration/index.spec.js [hunk 3]
```diff
@@ -92,6 +112,58 @@ describe('[INTEGRATION] index', function() {
...
+        var compiler = createCompiler(
+          {
+            tslintAutoFix: true,
+            tslint: true
+          },
+          false,
+          `./src/${fileName}.ts`
+        );
+        compiler.run(function(err, stats) {
+          expect(stats.compilation.warnings.length).to.be.eq(0);
```
Comment (piotr-oles) [inline test/integration/index.spec.js]: could you also check if content of the file is proper after fixing?

**Review by piotr-oles — CHANGES_REQUESTED** [review]

**Comment by ajainarayanan** [comment]
@piotr-oles @johnnyreilly Have updated the pull request. Can you give it another review.

**Comment by johnnyreilly** [comment]
Just merged in upstream changes @ajainarayanan.

@piotr-oles are you happy with the amends?

**Comment by ajainarayanan** [comment]
@johnnyreilly @piotr-oles I know its a little ridiculous to request some changes in an open source library and expect a release to happen right away but I am blocked by this change (or a new version of fork-ts-checker) to integrate prettier into my project.

Do we know tentatively when the pull request will be merged and a release will be cut? If any further changes are required please do let me know and I will make it.

**Comment by johnnyreilly** [comment]
Hey @ajainarayanan , sorry this is a bit of a frustration for you. From my perspective I'm fine for this to be merged and shipped now. Given @piotr-oles requested changes (which you've generously done!) I'd like him to confirm he's happy with the work rather than me merge directly.

If there's no response from him by the weekend then ping me again, maybe we can proceed without him. But that's not my preference.

Apologies for the delay; I understand the frustration.

**Comment by johnnyreilly** [comment]
Okay let's do this! Do you want to update the `package.json` and the `CHANGELOG.md`? Then I can cut a release.

**Comment by ajainarayanan** [comment]
@johnnyreilly Thanks! Have updated both. Could you give it a review?

**Comment by johnnyreilly** [comment]
Looks good - could you put the Default value of the field in the `README.md` please? Then we're good to go!

**Review by piotr-oles — APPROVED** [review]

**Comment by piotr-oles** [comment]
Thank you for your work and sorry for blocking this PR. Like @johnnyreilly said, please add default value to the README.md and then we're ready for release :)

**Comment by ajainarayanan** [comment]
@piotr-oles @johnnyreilly Updated. Thanks for the review.

**Comment by johnnyreilly** [comment]
Thanks! Should be available shortly

**Pull request merged**